app/code/Magento/SalesRule/Model/Rule/Metadata/ValueProvider.php

```diff
@@ -60,18 +30,13 @@ class ValueProvider
      * @param SimpleActionOptionsProvider|null $simpleActionOptionsProvider
      */
     public function __construct(
-        Store $store,
-        GroupRepositoryInterface $groupRepository,
-        SearchCriteriaBuilder $searchCriteriaBuilder,
-        DataObject $objectConverter,
-        RuleFactory $salesRuleFactory,
-        SimpleActionOptionsProvider $simpleActionOptionsProvider = null
+        protected readonly Store $store,
+        protected readonly GroupRepositoryInterface $groupRepository,
+        protected readonly SearchCriteriaBuilder $searchCriteriaBuilder,
+        protected readonly DataObject $objectConverter,
+        protected readonly RuleFactory $salesRuleFactory,
+        private ?SimpleActionOptionsProvider $simpleActionOptionsProvider = null
```

Comment:
The type of this argument seems to have changed by mistake.
```suggestion
        private SimpleActionOptionsProvider $simpleActionOptionsProvider = null
```